app/models/gallery.rb

```diff
@@ -75,12 +76,15 @@ def to_param
   def set_images_from_portal_urls
     transaction do
       enumerable_image_portal_urls.map { |url| Europeana::Record.id_from_portal_url(url) }.tap do |record_ids|
+        matched_ids = []
         record_ids.each_with_index do |record_id, i|
-          GalleryImage.find_or_create_by(gallery: self, europeana_record_id: record_id).tap do |image|
+          image_url = response_items.detect { |record| record['id'] == record_id }['edmIsShownBy'].first
```

Comment:
`edmIsShownBy` may be nil.

Reply:
not if the validation ran successfully, which happens before save, this whole method happens after save.
we could also set it to 'UNKNOWN' if we think it's going to be called outside of the normal cms or by some other script?

Reply:
Right you are. That's fine as-is then.